From aa9f8489fd8ff04eeefa6690101b90817f128dc5 Mon Sep 17 00:00:00 2001
From: Judith Mendez <jm@ti.com>
Date: Mon, 2 Feb 2026 13:59:27 -0600
Subject: [PATCH 5/7] PENDING: mmc: sdhci_am654: Add
 SDHCI_QUIRK_CAP_CLOCK_BASE_BROKEN quirk

Add quirk SDHCI_QUIRK_CAP_CLOCK_BASE_BROKEN to devices with DLL_PRESENT
so that DLL configuration can be determined using the actual clock to the
MMC card instead of the theoretical max clock.

Signed-off-by: Judith Mendez <jm@ti.com>
Tested-by: Moteen Shah <m-shah@ti.com>
---
 drivers/mmc/host/sdhci_am654.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/host/sdhci_am654.c b/drivers/mmc/host/sdhci_am654.c
index db94df574846..422f7a79d94f 100644
--- a/drivers/mmc/host/sdhci_am654.c
+++ b/drivers/mmc/host/sdhci_am654.c
@@ -669,7 +669,8 @@ static const struct sdhci_ops sdhci_j721e_8bit_ops = {
 
 static const struct sdhci_pltfm_data sdhci_j721e_8bit_pdata = {
 	.ops = &sdhci_j721e_8bit_ops,
-	.quirks = SDHCI_QUIRK_MULTIBLOCK_READ_ACMD12,
+	.quirks = SDHCI_QUIRK_MULTIBLOCK_READ_ACMD12 |
+		  SDHCI_QUIRK_CAP_CLOCK_BASE_BROKEN,
 	.quirks2 = SDHCI_QUIRK2_PRESET_VALUE_BROKEN |
 		   SDHCI_QUIRK2_DISABLE_HW_TIMEOUT,
 };
-- 
2.53.0

